sklearn/preprocessing/data.py

```diff
@@ -641,7 +643,14 @@ def partial_fit(self, X, y=None):
                 # First pass
                 if not hasattr(self, 'n_samples_seen_'):
                     self.mean_, self.var_ = mean_variance_axis(X, axis=0)
-                    self.n_samples_seen_ = X.shape[0]
+                    self.n_samples_seen_ = (np.ones(X.shape[1], dtype=np.int32)
+                                            * X.shape[0])
+                    sparse_constr = (sparse.csr_matrix if X.format == 'csr'
```

Comment:
I wish we could just use `X._with_data`

Reply:
Yep, it could be nice to have something like that publicly exposed.